### PR TITLE
Monitoring feature improvement

### PR DESCRIPTION
**Pull request by colin014**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| API breaks?     | no
| Deprecations?   | no
| License         | Apache 2.0

### What's in this PR?
Support new payload of the feature

### Checklist

- [x] Implementation tested (with at least one cloud provider)
- [x] Error handling code meets the [guideline](https://github.com/banzaicloud/pipeline/blob/master/docs/error-handling-guide.md)
- [x] Logging code meets the guideline (TODO)

### CODE REVIEW

File: internal/clusterfeature/features/monitoring/manager_test.go [hunk 1]
```diff
@@ -133,49 +137,172 @@ func TestFeatureManager_ValidateSpec(t *testing.T) {
...
+						"retention": "10m",
+					},
+					"ingress": obj{
 						"enabled": true,
 						"path":    prometheusPath,
 					},
 				},
+				"exporters": obj{
+					"enabled":          true,
+					"nodeExporter":     true,
```
Comment (sagikazarmark) [inline internal/clusterfeature/features/monitoring/manager_test.go]: Please check with @xyztdanid4 , but I believe the expected structure here is 
```json
{
    "nodeExporter": {
        "enabled": true
    }
}
```

File: internal/clusterfeature/features/monitoring/operator.go [hunk 2]
```diff
@@ -154,7 +207,7 @@ func (op FeatureOperator) Deactivate(ctx context.Context, clusterID uint, spec c
 		}
 	}
 
-	if boundSpec.Prometheus.SecretId == "" {
+	if boundSpec.Prometheus.Ingress.SecretId == "" {
```
Comment (sagikazarmark) [inline internal/clusterfeature/features/monitoring/operator.go]: This isn't gonna work. Generated secret IDs should be persisted in the spec.

File: internal/cluster/endpoints/common.go [hunk 3]
```diff
@@ -20,4 +20,5 @@ import (
 
 type EndpointService interface {
 	List(kubeConfig []byte, releaseName string) ([]*pkgHelm.EndpointItem, error)
+	GetServiceUrl(kubeConfig []byte, serviceName string, namespace string) (string, error)
```
[the remark this answers is not in the document]
Comment (lpuskas) [reply internal/cluster/endpoints/common.go]: i'd rename the List operation to ListEndpoints

File: internal/clusterfeature/features/monitoring/error.go [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+type requiredFieldError struct {
+	fieldName string
+}
+
+type invalidIngressHostError struct {
+	hostType string
+}
+
+type cannotDisabledError struct {
```
Comment (lpuskas) [inline internal/clusterfeature/features/monitoring/error.go]: rename this to `canNotDisableError`

File: internal/clusterfeature/features/monitoring/operator_secret.go [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+	secretRequest := &secret.CreateSecretRequest{
+		Name: m.infoer.generatedSecretName(),
+		Type: secrettype.HtpasswdSecretType,
+		Values: map[string]string{
+			secrettype.Username: generatedSecretUsername,
+			secrettype.Password: adminPass,
+		},
+		Tags: secretTags,
+	}
+	_, err = secret.Store.CreateOrUpdate(m.cluster.GetOrganizationId(), secretRequest)
```
[the remark this answers is not in the document]
Comment (colin014) [reply internal/clusterfeature/features/monitoring/operator_secret.go]: We discussed with @sagikazarmark earlier, and  `secret.Store` will be factor out from here in another PR

File: internal/clusterfeature/features/monitoring/operator.go (hunk 2 above)
Comment (colin014) [reply internal/clusterfeature/features/monitoring/operator.go]: this will be fixed in another PR

Comment (colin014) [reply internal/clusterfeature/features/monitoring/operator.go]: this will be fixed in another PR

**Pull request merged**